test/cluster/object_store/test_backup.py

```diff
@@ -407,6 +407,254 @@ async def test_abort_simple_backup_and_restore(manager: ManagerClient, object_st
+    # 8) write hundreds of keys, so data is spread across N*2 tablets.
+    # 9) after flushing, copy sstables somewhere
+    # The "copy sstables somewhere" steps are replaced by taking snapshots and backing them up to S3.
+
+    objconf = s3_storage.create_endpoint_conf()
+    config = {
+        'enable_user_defined_functions': False,
+        'object_storage_endpoints': objconf,
+        'experimental_features': ['keyspace-storage-options'],
+        'task_ttl_in_seconds': 300,
+    }
+    d = tmp_path / "system_keys"
+    d.mkdir()
+    config = config | {
+        'system_key_directory': str(d),
+        'user_info_encryption': { 'enabled': True, 'key_provider': 'LocalFileSystemKeyProviderFactory' }
+    }
+    cmd = ['--smp', '16','-m','32G', '--logger-log-level', 'sstables_loader=info:sstable=info']
+    servers = await manager.servers_add(servers_num=3, config=config, cmdline=cmd, auto_rack_dc="dc1")
+
+    # Obtain the CQL interface from the manager.
+    cql = manager.get_cql()
+
+    # Create keyspace, table, and fill data
+    print("Creating keyspace and table, then inserting data...")
+
+    initial_tablets = 8
+    target_tablets = initial_tablets * 2
+
+    # 1) create table with N tablets.
+    keyspace = 'test_ks'
+    table = 'test_cf'
+    await create_keyspace_and_table(manager, cql, servers, keyspace, table, initial_tablets)
+
+    # 2) disable balancing.
+    for server in servers:
+        # Disable auto compaction just to keep more sstables around for restore
+        await manager.api.disable_autocompaction(server.ip_addr, keyspace)
+        await manager.api.disable_tablet_balancing(server.ip_addr)
+
+    # 3) write hundreds of keys, so data is spread across N tablets.
+    insert_rows_mt(cql, keyspace, table, 100_000)
+
+    for server in servers:
+        await manager.api.flush_keyspace(server.ip_addr, keyspace)
+
+    row_count = 0
+    res = cql.execute(f"SELECT COUNT(*) FROM {keyspace}.{table} BYPASS CACHE USING TIMEOUT 600s;")
+    row_count += res[0].count
+    print(f"Initial row count: {row_count}")
+
+    # Take snapshot for keyspace
+    snapshot_name = unique_name('backup_')
+    print(f"Taking snapshot '{snapshot_name}' for keyspace '{keyspace}'...")
+    for server in servers:
+        await manager.api.take_snapshot(server.ip_addr, keyspace, snapshot_name)
+
+    # Collect snapshot files from each server
+    sstables = {
+        server.server_id: await get_snapshot_files(manager, server, keyspace, snapshot_name)
+        for server in servers
+    }
+    for server_id, toc_files in sstables.items():
+        print(f"Server ID: {server_id}, TOC files: {len(toc_files)}")
+
+    # Backup the keyspace on each server to S3
+    prefix = f"{table}/{snapshot_name}"
+    print(f"Backing up keyspace using prefix '{prefix}' on all servers...")
+    backup_tasks = {}
+    for server in servers:
+        backup_tasks[server.server_id] = await manager.api.backup(
```

Comment:
I think it makes sense running it in parallel using `asyncio.gather()` to (slightly) speed things up